rise/lib/location.py

```diff
@@ -0,0 +1,356 @@
+        for i in sorted(indices_to_pop, reverse=True):
+            copy_to_return.data.pop(i)
+
+        return copy_to_return
+
+    def filter_by_wkt(
+        self,
+        wkt: Optional[str] = None,
+        z: Optional[str] = None,
+    ):
+        """Filter a location by the well-known-text geometry representation"""
+        parsed_geo = shapely.wkt.loads(str(wkt)) if wkt else None
+        return self._filter_by_geometry(parsed_geo, z)
+
+    def drop_location(self, location_id: int):
+        """Given a location id, drop all all data that is associated with that location"""
+        new = self.model_copy()
+
+        filtered_locations = [
+            loc for loc in new.data if loc.attributes.id != location_id
+        ]
+
+        new.data = filtered_locations
+
+        return new
+
+    def filter_by_bbox(
+        self,
+        bbox: Optional[list] = None,
+        z: Optional[str] = None,
+    ):
+        if bbox:
+            parse_result = parse_bbox(bbox)
+            shapely_box = parse_result[0] if parse_result else None
+            z = parse_result[1] if parse_result else z
+
+        shapely_box = parse_bbox(bbox)[0] if bbox else None
+        # TODO what happens if they specify both a bbox with z and a z value?
+        z = parse_bbox(bbox)[1] if bbox else z
+
+        return self._filter_by_geometry(shapely_box, z)
+
+    def filter_by_limit(self, limit: int):
+        self.data = self.data[:limit]
+        return self
+
+    def remove_before_offset(self, offset: int):
+        self.data = self.data[offset:]
+        return self
+
+    def filter_by_id(
+        self,
+        identifier: Optional[str] = None,
+    ):
+        self.data = [
+            location
+            for location in self.data
+            if str(location.attributes.id) == identifier
+        ]
+        return self
+
+    def to_geojson(self, single_feature: bool = False) -> dict:
+        features = []
+
+        for location_feature in self.data:
+            # z = location_feature["attributes"]["elevation"]
+            # if z is not None:
+            #     location_feature["attributes"]["locationCoordinates"][
+            #         "coordinates"
+            #     ].append(float(z))
+            #     LOGGER.error(
+            #         location_feature["attributes"]["locationCoordinates"]["coordinates"]
+            #     )
+
+            feature_as_geojson = {
+                "type": "Feature",
+                "id": location_feature.attributes.id,
+                "properties": {
+                    "Locations@iot.count": 1,
+                    "name": location_feature.attributes.locationName,
+                    "id": location_feature.attributes.id,
+                    "Locations": [
+                        {
+                            "location": location_feature.attributes.locationCoordinates.model_dump()
+                        }
+                    ],
+                },
```

Comment:
We should not emulate STA GeoJSON here. Properties should basically be any of the `location_feature.attributes` we havent already mapped to GeoJSOn

Reply:
Not that im encouraging it, but surprised you aren't using https://github.com/developmentseed/geojson-pydantic